src/internal/connector/mockconnector/mock_data_list.go

```diff
@@ -1,16 +1,86 @@
 package mockconnector
 
 import (
+	"bytes"
+	"io"
+	"testing"
+
 	kw "github.com/microsoft/kiota-serialization-json-go"
 	"github.com/microsoftgraph/msgraph-sdk-go/models"
+	"github.com/stretchr/testify/require"
+
+	"github.com/alcionai/corso/src/internal/data"
+	"github.com/alcionai/corso/src/pkg/path"
+)
+
+var (
+	_ data.Stream     = &MockListData{}
+	_ data.Collection = &MockListCollection{}
 )
 
+type MockListCollection struct {
+	fullPath path.Path
+	Data     []*MockListData
+	Names    []string
+}
+
+func (mlc *MockListCollection) State() data.CollectionState {
+	return data.NewState
+}
+
+func (mlc *MockListCollection) FullPath() path.Path {
+	return mlc.fullPath
+}
+
+func (mlc *MockListCollection) DoNotMergeItems() bool {
+	return false
+}
+
+func (mlc *MockListCollection) PreviousPath() path.Path {
+	return mlc.fullPath
```

Comment:
This could have unexpected side effects.  Better for it to be nil until there's an actual previousPath.  This will help ensure that if/when incrementals do get picked up for sharepoint, these mocks always treat their data as being in a "new" state (ie: not an incremental update), as per the expectations of the State field.
```suggestion
	return nil
```